lib/senders/email.js

```diff
@@ -253,97 +267,227 @@ module.exports = function (log, config) {
 
-    if (this.sesConfigurationSet) {
-      // Note on SES Event Publishing: The X-SES-CONFIGURATION-SET and
-      // X-SES-MESSAGE-TAGS email headers will be stripped by SES from the
-      // actual outgoing email messages.
-      headers[X_SES_CONFIGURATION_SET] = this.sesConfigurationSet
-      headers[X_SES_MESSAGE_TAGS] = sesMessageTagsHeaderValue(message.metricsTemplate || template, emailService)
-    }
+          // Set headers that let us attribute success/failure correctly
+          message.emailService = headers['X-Email-Service'] = emailService
+          message.emailSender = headers['X-Email-Sender'] = emailSender
 
-    log.info({
-      email: message.email,
-      op: 'mailer.send',
-      template,
-      headers: Object.keys(headers).join(',')
-    })
+          if (this.sesConfigurationSet && emailSender === 'ses') {
+            // Note on SES Event Publishing: The X-SES-CONFIGURATION-SET and
+            // X-SES-MESSAGE-TAGS email headers will be stripped by SES from the
+            // actual outgoing email messages.
+            headers[X_SES_CONFIGURATION_SET] = this.sesConfigurationSet
+            headers[X_SES_MESSAGE_TAGS] = sesMessageTagsHeaderValue(message.metricsTemplate || template, emailService)
+          }
 
-    var emailConfig = {
-      sender: this.sender,
-      from: this.sender,
-      to: message.email,
-      subject: localized.subject,
-      text: localized.text,
-      html: localized.html,
-      xMailer: false,
-      headers
-    }
+          log.info({
+            email: emailAddresses[0],
+            op: 'mailer.send',
+            template,
+            headers: Object.keys(headers).join(',')
+          })
+
+          const emailConfig = {
+            sender: this.sender,
+            from: this.sender,
+            to: emailAddresses[0],
+            subject: localized.subject,
+            text: localized.text,
+            html: localized.html,
+            xMailer: false,
+            headers
+          }
 
-    // Utilize nodemailer's cc ability to send to multiple addresses
-    // Ref: https://nodemailer.com/message/
-    if (message.ccEmails) {
-      emailConfig.cc = message.ccEmails
-    }
+          if (emailAddresses.length > 1) {
+            emailConfig.cc = emailAddresses.slice(1)
+          }
 
-    var d = P.defer()
-    mailer.sendMail(
-      emailConfig,
-      function (err, status) {
-        if (err) {
-          log.error(
-            {
+          const d = P.defer()
+          mailer.sendMail(emailConfig, (err, status) => {
+            if (err) {
+              log.error({
+                op: 'mailer.send.1',
+                err: err && err.message,
+                status: status && status.message,
+                id: status && status.messageId,
+                to: emailConfig && emailConfig.to
+              })
+
+              return d.reject(err)
+            }
+
+            log.info({
```

Comment:
Not related to this PR - would we ever care which sender sent the message? Can we determine that from the log entry?

Reply:
I've added `emailSender` and `emailService` to these log messages.